sktime/forecasting/prophet.py

```diff
@@ -0,0 +1,341 @@
+
+    """
+
+    def __init__(
+        self,
+        # Args due to wrapping
+        freq,
+        add_seasonality=None,
+        add_country_holidays=None,
+        # Args of fbprophet
+        growth='linear',
+        changepoints=None,
+        n_changepoints=25,
+        changepoint_range=0.8,
+        yearly_seasonality='auto',
+        weekly_seasonality='auto',
+        daily_seasonality='auto',
+        holidays=None,
+        seasonality_mode='additive',
+        seasonality_prior_scale=10.0,
+        holidays_prior_scale=10.0,
+        changepoint_prior_scale=0.05,
+        mcmc_samples=0,
+        interval_width=1 - DEFAULT_ALPHA,
+        uncertainty_samples=1000,
+        stan_backend=None,
+        **kwargs
+    ):
+        self.freq = freq
+        self.add_seasonality = add_seasonality
+        self.add_country_holidays = add_country_holidays
+
+        self.growth = growth
+        self.changepoints = changepoints
+        if self.changepoints is not None:
+            self.changepoints = pd.Series(pd.to_datetime(self.changepoints), name='ds')
+            self.n_changepoints = len(self.changepoints)
+            self.specified_changepoints = True
+        else:
+            self.n_changepoints = n_changepoints
+            self.specified_changepoints = False
+
+        self.changepoint_range = changepoint_range
+        self.yearly_seasonality = yearly_seasonality
+        self.weekly_seasonality = weekly_seasonality
+        self.daily_seasonality = daily_seasonality
+        self.holidays = holidays
+
+        self.seasonality_mode = seasonality_mode
+        self.seasonality_prior_scale = float(seasonality_prior_scale)
+        self.changepoint_prior_scale = float(changepoint_prior_scale)
+        self.holidays_prior_scale = float(holidays_prior_scale)
+
+        self.mcmc_samples = mcmc_samples
+        self.interval_width = interval_width
+        self.uncertainty_samples = uncertainty_samples
+
+        self.stan_backend = stan_backend
+
+        super(Prophet, self).__init__()
+
+        # import inside method to avoid hard dependency
+        from fbprophet.forecaster import Prophet as _Prophet
+
+        self._forecaster = _Prophet(
+            growth=growth,
+            changepoints=changepoints,
+            n_changepoints=n_changepoints,
+            changepoint_range=changepoint_range,
+            yearly_seasonality=yearly_seasonality,
+            weekly_seasonality=weekly_seasonality,
+            daily_seasonality=daily_seasonality,
+            holidays=holidays,
+            seasonality_mode=seasonality_mode,
+            seasonality_prior_scale=seasonality_prior_scale,
+            holidays_prior_scale=holidays_prior_scale,
+            changepoint_prior_scale=changepoint_prior_scale,
+            mcmc_samples=mcmc_samples,
+            interval_width=interval_width,
+            uncertainty_samples=uncertainty_samples,
+            stan_backend=stan_backend,
+            **kwargs
+        )
```

Comment:
I think I'd move this to the beginning of `fit` and pass on the values from `self`, otherwise values may not be passed on correctly if I use `set_params()` instead of re-instantiating the object.

Reply:
I actually dont understand this point, in https://github.com/alan-turing-institute/sktime/blob/4422a66a2749eaea41d3fbda723bc8c64f334787/sktime/forecasting/arima.py#L306-L310 it is also done like I did it?

Reply:
It should be in fit there too I believe. I worry about the following situation:
```python

from sktime.forecasting.all import *

f = AutoARIMA()
f._forecaster.get_params()["max_d"]
>>> 2  # default value
f.set_params(**{"max_d": 1})
f._forecaster.get_params()["max_d"]
>>> 2  # internal value not updated 

y = load_airline()
f.fit(y)
f._forecaster.get_params()["max_d"]
>>> 2  # the wrong value is used in fit 
```